vp/devel/headsail.repl

```diff
@@ -0,0 +1,277 @@
+    allowUnalignedAccesses: false
+
+/*
+ * CLINT [^1]
+ *
+ * [IP](https://gitlab.tuni.fi/soc-hub/common/hw/cva6/-/tree/master/corev_apu/clint)
+ * [Specification](https://github.com/pulp-platform/clint) (unconfirmed)
+ */
+clint: IRQControllers.CoreLevelInterruptor @ sysbus 0x100060000
+    [0, 1] -> cpu_hpc0@[3, 7] // msoft, mtimer
+    [2, 3] -> cpu_hpc1@[3, 7] // msoft, mtimer
+    [4, 5] -> cpu_hpc2@[3, 7] // msoft, mtimer
+    [6, 7] -> cpu_hpc3@[3, 7] // msoft, mtimer
+    numberOfTargets: 4
+    frequency: 32768 // we guess the usual RTC freq. of 32.768 kHz
+
+/*
+ * PLIC [^1]
+ * * [IP](https://github.com/pulp-platform/rv_plic)
+ * [Specification](https://github.com/riscv/riscv-plic-spec/blob/master/riscv-plic.adoc)
+ *
+ * Some addresses were tightened to save space -> ASIC impl will differ from sim.
+ * Length on AISC: 0x1_0000
+ * Length on VP: 0x4000_000
+ * This takes memory from HPC_SDRAM
+ */
+plic: IRQControllers.PlatformLevelInterruptController @ sysbus 0x100080000
+    [0, 1] -> cpu_hpc0@[11, 9] // mext, sext
+    [2, 3] -> cpu_hpc1@[11, 9] // mext, sext
+    [4, 5] -> cpu_hpc2@[11, 9] // mext, sext
+    [6, 7] -> cpu_hpc3@[11, 9] // mext, sext
+    numberOfSources: 23
+    numberOfContexts: 8
+
+/*
+ * PULP APB timer [^1]
+ * [IP](https://github.com/pulp-platform/apb_timer)
+ *
+ * Interrupts are controlled by PLIC.
+ */
+timer: Timers.PULP_Timer @ sysbus <0x100050000, +0x1000>
+    /* Renode currently hardcodes the number of timers to 2, while Headsail specifies 4
+     * See also: https://github.com/renode/renode-infrastructure/blob/579687e750c3da24a362e899d312f4e738209ae7/src/Emulator/Peripherals/Peripherals/Timers/PULP_Timer.cs#L146
+    // [ovf, cmp, ovf, cmp, ...]
+    [0, 1, 2, 3, 4, 5, 6, 7] -> plic@[1, 2, 3, 4, 5, 6, 7, 8]
+    */
+    // [ovf, cmp]
+    [0, 1] -> plic@[1, 2]
+    frequency: 100000 // we guess board reference frequency
+
+mem_dsp_0: Memory.MappedMemory @ {
+    sysbus new Bus.BusPointRegistration {address: 0xFF810000; cpu: cpu_sysctrl};
+    sysbus new Bus.BusPointRegistration {address: 0x1FF810000; cpu: cpu_hpc0};
+    sysbus new Bus.BusPointRegistration {address: 0x1FF810000; cpu: cpu_hpc1};
+    sysbus new Bus.BusPointRegistration {address: 0x1FF810000; cpu: cpu_hpc2};
+    sysbus new Bus.BusPointRegistration {address: 0x1FF810000; cpu: cpu_hpc3}
+    }
+    size: 0x10000
+
+mem_dsp_1: Memory.MappedMemory @ {
+    sysbus new Bus.BusPointRegistration {address: 0xFF830000; cpu: cpu_sysctrl};
+    sysbus new Bus.BusPointRegistration {address: 0x1FF830000; cpu: cpu_hpc0};
+    sysbus new Bus.BusPointRegistration {address: 0x1FF830000; cpu: cpu_hpc1};
+    sysbus new Bus.BusPointRegistration {address: 0x1FF830000; cpu: cpu_hpc2};
+    sysbus new Bus.BusPointRegistration {address: 0x1FF830000; cpu: cpu_hpc3}
+    }
+    size: 0x10000
+
+/*
+ * NOTE: Boot ram is included in SDRAM hpc
+ */
+hpc_bootram: Memory.MappedMemory @ {
+    sysbus new Bus.BusPointRegistration {address: 0x10000; cpu: cpu_sysctrl};
```

Comment:
Should be visible also to HPC in 0x1_0000.

Reply:
Bootram is now visible in 0x1_0000 for all HPC cores